### PR TITLE
Fix for nv-bug 4786175

### PR DESCRIPTION
**Pull request by zoobereq**
# What does this PR do ?

Fixes [nv-bug #4786175](https://nvbugspro.nvidia.com/bug/4786175).

The fix implements `per_unit` mappings in lieu of morphosyntactic features.  The list of mappings is contained in `nemo_text_processing/text_normalization/en/data/money/per_unit.tsv` and may be expanded to include additional mappings.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Have you signed your commits? Use ``git commit -s`` to sign.
- [x] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [x] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [x] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [x] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [x] Have you added the correct license header `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [x] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [x] Remove import guards (`try import: ... except: ...`) if not already done.
- [x] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [x] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation
- [ ] Test

If you haven't finished some of the above items you can still open "Draft" PR.

### CODE REVIEW

File: nemo_text_processing/text_normalization/en/taggers/money.py [hunk 1]
```diff
@@ -183,11 +182,16 @@ def __init__(self, cardinal: GraphFst, decimal: GraphFst, deterministic: bool =
...
+
         if not deterministic:
             final_graph |= integer_graph_reordered | decimal_default_reordered
             # to handle "$2.00" cases
             final_graph |= pynini.compose(
-                NEMO_SIGMA + pynutil.delete(".") + pynini.closure(pynutil.delete("0"), 1), integer_graph_reordered
+                NEMO_SIGMA + pynutil.delete(".") + pynini.closure(pynutil.delete("0"), 1), integer_graph_reordered,
             )
+
+        final_graph += graph_per_units.ques
```
Comment (tbartley94) [inline nemo_text_processing/text_normalization/en/taggers/money.py:195]: You want it recursive `per month per quarter`

**Review by ekmb — APPROVED** [review]

**Review by tbartley94 — CHANGES_REQUESTED** [review]
To be semantically valid, this should be a features of `measures` that imports the money tagger as a nested class. (I believe this sparrowhawk safe). So for a merge to main, I would want to place it there for more generalizability.

Since it's a customer request, it probably would work better to provide them a separate fork for a quick fix and then we can do the proper merge into measures later.

**Comment by tbartley94** [comment]
@ekmb Please revert per above comment

**Comment by zoobereq** [comment]
_"To be semantically valid, this should be a features of measures that imports the money tagger as a nested class. (I believe this sparrowhawk safe). So for a merge to main, I would want to place it there for more generalizability."_

I agree, and this was in fact my first approach to this issue (see the comment on [this Jira](https://jirasw.nvidia.com/secure/RapidBoard.jspa?rapidView=29211&projectKey=JPE&view=detail&selectedIssue=JPE-8179&quickFilter=87305#) from 08/19).  While `pynini/pyTest` works, SH breaks.  Looking at the SH proto for semiotic classes, it doesn't look like `MONEY` can be passed into it ([link](https://github.com/yzhang123/sparrowhawk/blob/test/src/proto/semiotic_classes.proto#L137)), but I may of course be missing something?  Please let me know.

_"Since it's a customer request, it probably would work better to provide them a separate fork for a quick fix and then we can do the proper merge into measures later."_

The deadline for this is tomorrow.  Please let me know the best way to handle it for the customer.

File: nemo_text_processing/text_normalization/en/taggers/money.py (hunk 1 above)
Comment (zoobereq) [reply nemo_text_processing/text_normalization/en/taggers/money.py:195]: Good point. Let me work on that.

**Pull request merged**